agents/builtin/mysql/mysql.go

```diff
@@ -93,78 +114,179 @@ func (m *MySQL) Run(ctx context.Context) {
-		   * report rows with NULL digest?
-		   * get query by digest from events_statements_history_long?
-		   * check/report the value of performance_schema_events_statements_history_long_size?
-		   * set conditions for FIRST_SEEN / LAST_SEEN? what conditions?
-		   * group/aggregate results? how?
-		   * should github.com/percona/go-mysql/event be used?
-		*/
-
-		res.MetricsBucket = append(res.MetricsBucket, &qanpb.MetricsBucket{
-			Queryid:     *ess.Digest,
-			Fingerprint: *ess.DigestText,
-			DServer:     "TODO",
-			DDatabase:   "TODO",
-			DSchema:     "TODO",
-		})
+		for _, p := range []struct {
+			value float32  // result value: currentESS.SumXXX-prevESS.SumXXX
+			sum   *float32 // MetricsBucket.XXXSum field to write value
+			cnt   *float32 // MetricsBucket.XXXCnt field to write count
+		}{
+			// in order of events_statements_summary_by_digest columns
+
+			// convert picoseconds to seconds
+			{float32(currentESS.SumTimerWait-prevESS.SumTimerWait) / 1e12, &mb.MQueryTimeSum, &mb.MQueryTimeCnt},
+			{float32(currentESS.SumLockTime-prevESS.SumLockTime) / 1e12, &mb.MLockTimeSum, &mb.MLockTimeCnt},
+
+			{float32(currentESS.SumRowsAffected - prevESS.SumRowsAffected), &mb.MRowsAffectedSum, &mb.MRowsAffectedCnt},
+			{float32(currentESS.SumRowsSent - prevESS.SumRowsSent), &mb.MRowsSentSum, &mb.MRowsSentCnt},
+			{float32(currentESS.SumRowsExamined - prevESS.SumRowsExamined), &mb.MRowsExaminedSum, &mb.MRowsExaminedCnt},
+
+			{float32(currentESS.SumCreatedTmpDiskTables - prevESS.SumCreatedTmpDiskTables), &mb.MTmpDiskTablesSum, &mb.MTmpDiskTablesCnt},
+			{float32(currentESS.SumCreatedTmpTables - prevESS.SumCreatedTmpTables), &mb.MQueryTimeSum, &mb.MTmpTableCnt},
+			{float32(currentESS.SumSelectFullJoin - prevESS.SumSelectFullJoin), &mb.MFullJoinSum, &mb.MFullJoinCnt},
+			{float32(currentESS.SumSelectFullRangeJoin - prevESS.SumSelectFullRangeJoin), &mb.MSelectFullRangeJoinSum, &mb.MSelectFullRangeJoinCnt},
+			{float32(currentESS.SumSelectRange - prevESS.SumSelectRange), &mb.MSelectRangeSum, &mb.MSelectRangeCnt},
+			{float32(currentESS.SumSelectRangeCheck - prevESS.SumSelectRangeCheck), &mb.MSelectRangeCheckSum, &mb.MSelectRangeCheckCnt},
+			{float32(currentESS.SumSelectScan - prevESS.SumSelectScan), &mb.MFullScanSum, &mb.MFullScanCnt},
+
+			{float32(currentESS.SumSortMergePasses - prevESS.SumSortMergePasses), &mb.MMergePassesSum, &mb.MMergePassesCnt},
+			{float32(currentESS.SumSortRange - prevESS.SumSortRange), &mb.MSortRangeSum, &mb.MSortRangeCnt},
+			{float32(currentESS.SumSortRows - prevESS.SumSortRows), &mb.MSortRowsSum, &mb.MSortRowsCnt},
+			{float32(currentESS.SumSortScan - prevESS.SumSortScan), &mb.MSortScanSum, &mb.MSortScanCnt},
+
+			{float32(currentESS.SumNoIndexUsed - prevESS.SumNoIndexUsed), &mb.MNoIndexUsedSum, &mb.MNoIndexUsedCnt},
+			{float32(currentESS.SumNoGoodIndexUsed - prevESS.SumNoGoodIndexUsed), &mb.MNoGoodIndexUsedSum, &mb.MNoGoodIndexUsedCnt},
+		} {
+			if p.value != 0 {
+				*p.sum = p.value
```

Comment:
IMHO, `p` should have more descriptive name here, to easier catch it purposes.

Reply:
`p` is a local loop variable, it has to be short. And I can't think of a better name.